htdocs/mri/jiv/get_file.php

```diff
@@ -19,6 +19,46 @@
  *  @link     https://github.com/aces/Loris-Trunk
  */
 
+/**
+ * This function is to replace PHP's extremely buggy realpath().
+ * Taken from https://gist.github.com/umidjons/6154548
+ *
+ * @param string $path The original path, can be relative etc.
+ *
+ * @return string The resolved path, it might not exist.
+ */
+function truepath( $path )
+{
+    // whether $path is unix or not
+    $unipath = strlen($path) == 0 || $path{0} != '/';
+    // attempts to detect if path is relative in which case, add cwd
+    if (strpos($path, ':') === false && $unipath ) {
+        $path = getcwd() . DIRECTORY_SEPARATOR . $path;
+    }
+    // resolve path parts (single dot, double dot and double delimiters)
+    $path      = str_replace(array( '/', '\\' ), DIRECTORY_SEPARATOR, $path);
+    $parts     = array_filter(explode(DIRECTORY_SEPARATOR, $path), 'strlen');
+    $absolutes = array();
+    foreach ( $parts as $part ) {
+        if ('.' == $part ) {
+            continue;
+        }
+        if ('..' == $part ) {
+            array_pop($absolutes);
```

Comment:
It might be a good idea to check the return value here and throw an exception if you're trying to pop off of an empty stack.